### PR TITLE
Disable GPU access when platform view is destroyed

### PR DESCRIPTION
**Pull request by dnfield**
This fixes https://github.com/flutter/flutter/issues/73675 locally.

I suspect this will help with https://github.com/flutter/flutter/issues/64558 as well, though I cannot reproduce that one locally to test it.

Basic idea is that when the PlatformView is destroyed, disable the IO and Raster GPU access, and then re-enable them when the view is created again.

### CODE REVIEW

File: shell/platform/android/platform_view_android.cc [hunk 1]
```diff
@@ -414,6 +414,7 @@ void PlatformViewAndroid::InstallFirstFrameCallback() {
 
 void PlatformViewAndroid::FireFirstFrameCallback() {
   jni_facade_->FlutterViewOnFirstFrame();
+  delegate_.GetIsGpuDisabledSyncSwitch()->SetSwitch(true);
```
Comment (dnfield) [inline shell/platform/android/platform_view_android.cc:417]: May be splittable to another patch.

Comment (dnfield) [reply shell/platform/android/platform_view_android.cc:417]: Need to do something similar on other platforms if we make the change in shell.cc as part of this patch.

**Comment by dnfield** [comment]
I've updated this with what I think is a better solution, and it's ready for review. This definitely fixes the Android bug, not sure about iOS because I can't repro that one locally.

File: shell/common/shell_io_manager.cc [hunk 2]
```diff
@@ -79,8 +79,7 @@ ShellIOManager::ShellIOManager(
 ShellIOManager::~ShellIOManager() {
   // Last chance to drain the IO queue as the platform side reference to the
   // underlying OpenGL context may be going away.
-  is_gpu_disabled_sync_switch_->Execute(
-      fml::SyncSwitch::Handlers().SetIfFalse([&] { unref_queue_->Drain(); }));
+  unref_queue_->Drain();
```
Comment (dnfield) [inline shell/common/shell_io_manager.cc:82]: @gaaclarke - I think this is not safe on iOS, particularly if thereare multiple engines. Dunno what you think of maybe guarding this specifically for iOS.

**Comment by Sulisong** [comment]
Thanks to the developers! I look forward to using the beta version as soon as possible. Happy travel for developers

File: shell/common/shell.cc [hunk 3]
```diff
@@ -699,6 +699,8 @@ void Shell::OnPlatformViewCreated(std::unique_ptr<Surface> surface) {
   FML_DCHECK(is_setup_);
   FML_DCHECK(task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread());
 
+  GetIsGpuDisabledSyncSwitch()->SetSwitch(false);
```
Comment (gaaclarke) [inline shell/common/shell.cc:702]: This isn't how this API was intended to be used.  The setting of the switch was intended to happen when the app loses access to the GPU.  The calls to turn it off and on are matched calls provided by the operating system.  I think this could get really messy fixing our problems this way.

**Review by gaaclarke — COMMENTED** [review]
Sorry, just getting time now to truly consider what is happening in the PR.

File: shell/common/shell.cc (hunk 3 above)
Comment (dnfield) [reply shell/common/shell.cc:702]: I'm going to try to rework this a bit.

**Comment by dnfield** [comment]
I think I have a better approach for this - going to close this and expect to open a new one later today.
